### ISSUE TITLE
ERROR 2002 (HY000): Can't connect to MySQL server on 'db' (115)

### ISSUE THREAD

**Issue by ahsan-horani-folio** [markshust/docker-magento#589]
I am trying to move existing project to Docker. I followed your tutorial at [M.Academy](https://courses.m.academy/courses/set-up-magento-2-development-environment-docker/lectures/9283467 ) and followed the setup instructions from [Docker on Existing Project](https://github.com/markshust/docker-magento#existing-projects)

While executing `bin/mysql < /var/www/magento243.sql` I am getting this error ERROR 2002 (HY000): Can't connect to MySQL server on 'db' (115)

I also tried to execute: `bin/root telnet db 3306`

Response:
Trying 172.17.0.1...
telnet: Unable to connect to remote host: Connection timed out

I tried everything but couldn't figure out the problem. I am new to Docker. I am using Ubuntu 18.04 LTE. Haven't changed db.env and got no error on any other step before importing the database.

I further checked and found out that the container is not able to establish connection with MySQL. Try `bin/mysql` and the same error will be returned.

P.S. I am successfully able to connect with same MySQL service outside Docker container by using `mysql -h 127.0.01 -u root -p -P 3306`

**Steps To Reproduce**
1. Install Docker & Docker Compose on Ubuntu 18.04
2. Download the Docker Compose template: `curl -s https://raw.githubusercontent.com/markshust/docker-magento/master/lib/template | bash`
3. Replace with existing source code of your existing Magento instance: `cp -R ~/Sites/existing src`
4. Execute: `docker-compose -f docker-compose.yml up -d`
5. Copy files to container: `bin/copytocontainer --all`
6. Import existing database: `bin/mysql < /var/www/magento243.sql`

**Expected Result**
Database should have been successfully imported

**Actual Result**
ERROR 2002 (HY000): Can't connect to MySQL server on 'db' (115)

**Comment by ahsan-horani-folio** [comment]
Any update on this? @markhust

**Comment by markshust** [comment]
@ahsan-horani-folio I'm unable to replicate this.

**Comment by ahsan-horani-folio** [comment]
What could be the potential issue here? Server specific? Any blocked port?

**Comment by markshust** [comment]
If you are inside the container, there is no blocked port to deal with. So, when you enter the container with `bin/bash` you are within the Docker network. If you can't connect via the bash shell, either the MySQL credentials were changed or modified, or the MySQL/`db` service is not running.

**Comment by lcailleux** [comment]
I had the same problem on Linux, and I was able to make it work by replacing some of the M1 mac instructions which were set by default in the docker-compose.yml file.

`- "db:172.17.0.1"`

I changed the IP address by the result of this command:

`docker run --rm alpine ip route | awk 'NR==1 {print $3}'`

Not sure if those instructions are even needed for Linux users. I hope this helps.

**Comment by markshust** [comment]
@lcailleux yep, these were added for M1 support https://github.com/markshust/docker-magento/blob/master/compose/docker-compose.yml#L32 otherwise long lags would happen with the DNS lookups. I didn't know this happened on Linux as well!

**Comment by sr-expert** [comment]
i have same error

**Comment by r-martins** [comment]
Commenting `      - "db:172.17.0.1"` will fix the issue (at least for those mac users _not_ using M1 Mac).

**Comment by markshust** [comment]
@r-martins this host entry should already exist, perhaps you're on an older install. See https://github.com/markshust/docker-magento/blob/master/compose/docker-compose.yml#L32

**Comment by markshust** [comment]
@r-martins just realized you said to comment out that line. Can anyone else confirm commenting out that line on Linux fixes your issue?

I'm wondering if this ticket https://github.com/markshust/docker-magento/issues/636 will resolve this and some other related issues, as this would allow us to get rid of all of those host entries everyplace.

**Comment by ikslavok** [comment]
> @r-martins just realized you said to comment out that line. Can anyone else confirm commenting out that line on Linux fixes your issue?
> 
> I'm wondering if this ticket #636 will resolve this and some other related issues, as this would allow us to get rid of all of those host entries everyplace.

Commenting out that line solves the issue for me too.

**Comment by sr-expert** [comment]
my soluton for linux is =>
docker-compose-dev.yml

version: "3"

services:
  app:
    volumes: &appvolumes
      - ./src/nginx.conf.sample:/var/www/html/nginx.conf:cached
      - ./src:/var/www/html:cached
      - ./env/php.ini:/usr/local/etc/php/php.ini

  phpfpm:
    volumes: *appvolumes
    ## Replace 172.17.0.1 with the result of:
    ## docker run --rm alpine ip route | awk 'NR==1 {print $3}'
    extra_hosts:
      - "host.docker.internal:10.200.0.1"
and 

docker-compose.yml =>

change 172.17.0.1 => Replace 172.17.0.1 with the result of:docker run --rm alpine ip route | awk 'NR==1 {print $3}'


and is work for me

**Comment by markshust** [comment]
I'm going to close this issue out in favor of https://github.com/markshust/docker-magento/issues/636 -- I have tested new Bullseye images, and they all work ok, which completely eliminates the need to use extra_hosts directives on Mac (and probably Linux at this point).
